source/postprocess/particle_pdf.cc

```diff
@@ -0,0 +1,276 @@
+                {
+                  if (dim == 3)
+                    {
+                      for (unsigned int z = 0; z< pdf_granularity; ++z)
+                        {
+                          const double this_value = evaluate_function_at_index(x,y,z);
+                          max = std::max(max, this_value);
+                          min = std::min(min, this_value);
+                          // Sum in mean, then divide after this loop
+                          mean += this_value;
+                        }
+                    }
+                  else
+                    {
+                      const double this_value = evaluate_function_at_index(x,y,0);
+                      max = std::max(max, this_value);
+                      min = std::min(min, this_value);
+                      // Sum in mean, then divide after this loop
+                      mean += this_value;
+                    }
+                }
+            }
+          // Set the true mean
+          mean /= Utilities::fixed_power<dim>(pdf_granularity);
+          double squared_deviation_sum = 0;
+
+          // Then sum all the squared deviations for standard deviation
+          for (unsigned int x = 0; x< pdf_granularity; ++x)
+            {
+              for (unsigned int y = 0; y< pdf_granularity; ++y)
+                {
+                  if (dim==3)
+                    {
+                      for (unsigned int z = 0; z< pdf_granularity; ++z)
+                        {
+                          TableIndices<dim> entry_index;
+                          entry_index[0] = x;
+                          entry_index[1] = y;
+                          entry_index[2] = z;
+                          const double this_value = function_output_table(entry_index);
+                          const double deviation_squared = (this_value-mean)*(this_value-mean);
+                          squared_deviation_sum += deviation_squared;
+                        }
+                    }
+                  else
+                    {
+                      TableIndices<dim> entry_index;
+                      entry_index[0] = x;
+                      entry_index[1] = y;
+                      const double this_value = function_output_table(entry_index);
+                      const double deviation_squared = (this_value-mean)*(this_value-mean);
+                      squared_deviation_sum += deviation_squared;
+                    }
+                }
+            }
+          // Standard deviation of all the defined points in the density function
+          squared_deviation_sum /= (pdf_granularity*dim);
+          standard_deviation = std::sqrt(squared_deviation_sum);
+        }
+      else
+        {
+          // Loop through the vector and set min, max, sum mean.
+          for (const double this_value : function_output_vector)
+            {
+              max = std::max(max, this_value);
+              min = std::min(min, this_value);
```

Comment:
but didnt you set min and max for this case already when creating the function_output_vector? I think you can remove this here.